rest-client/src/main/java/io/apicurio/registry/client/request/AuthInterceptor.java

```diff
@@ -39,7 +43,19 @@ public AuthInterceptor(Auth auth) {
     public Response intercept(Chain chain) throws IOException {
         final Request request = chain.request();
         Map<String, String> headers = new HashMap<>();
-        auth.apply(headers);
+        try {
+            auth.apply(headers);
```

Comment:
I don't see how this can throw a NotAuthorizedException from jax-rs.

Reply:
keycloak client does